### PR TITLE
Added testsuite for metrics

### LINKED ISSUE #44: Test Metrics
Let's write some test code to make sure that the metrics work as expected. Let's give the metric functions simple data with known values and make sure they are behaving as expected.

### PR DESCRIPTION
**Pull request by orionw**
- All metrics tested

Note: found slight bug in spearman.  However, it should flip our spearman correlation (so instead of -0.003 we should be 0.003).  So there is a bug, but it does not fix all our problems. Let's talk about it and fix it in a separate PR.

Closes #44

### CODE REVIEW

File: dna/metrics.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import scipy.stats
 from sklearn.metrics import accuracy_score, mean_squared_error
 
-import utils
+from dna import utils
```
Comment (orionw) [inline dna/metrics.py]: This is what will cause the regular DNA to fail, but without it the testsuite will fail.

**Comment by bjschoenfeld** [comment]
@erikhuck ran into the same problem trying to run a test suite and dna.__main__.py. He is trying to turn dna into a full python package with a setup.py. Perhaps installing the package locally will fix the issue. We will see.

**Comment by bjschoenfeld** [comment]
I fixed the issue that was causing a conflict with main and the tests.

There is a lot of duplication of code in the tests which is hard to maintain. You can use `setUp`, `tearDown`, and custom auxiliary methods to help, see [here](https://docs.python.org/3.6/library/unittest.html#organizing-tests).

File: tests/test_metrics.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        ranked_data, actual_data = self.format_top_k(scores, ids, rank)
+        metric = top_k_function(ranked_data, actual_data, k)
+        return metric
+
+    def format_and_get_spearman(self, actual_data, ranked_data):
+        """
+        :param actual_data: a list of the real rankings
+        :param ranked_data: a list of the ranked predictions
+        """
+        actual_data.reverse() # our rank function in spearman will reverse this back
```
Comment (bjschoenfeld) [inline tests/test_metrics.py]: If our spearman function is off by a sign, this seems like it hides the problem rather than fixing it. Is that what this does?

Comment (orionw) [reply tests/test_metrics.py]: Yes, we do need to fix the spearman function.  But how we fix it will decide what we need to do here.  See my comment in the PR opening
> Let's talk about it and fix it in a separate PR.

**Comment by orionw** [comment]
I'm not sure what can be de-duplicated - if you have any specific examples, please let me know.

File: tests/test_metrics.py (hunk 2 above)
Comment (bjschoenfeld) [reply tests/test_metrics.py]: It's better for the tests to be correct and fail, then to hide a bug in our code. That is the point. We can still fix it in a separate PR if you wish.
